### PR TITLE
Remove querier wait time metric.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
We would like to know how long a querier worker is idle to understand if workstealing would have an impact. The original metric was too noisy and its cardinality was too high. Instead, we are going to log the wait time.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/querier/worker/processor_manager.go [hunk 1]
```diff
@@ -64,7 +65,9 @@ func (pm *processorManager) concurrency(n int) {
 		n = 0
 	}
 
+	workerId := 0
```
Comment (ashwanthgoli) [inline pkg/querier/worker/processor_manager.go:68]: i am guessing the workerId does not matter much here. but there is a chance here of re-using workerIds if the concurrency value changes.

Comment (jeschkies) [reply pkg/querier/worker/processor_manager.go:68]: Good question. I wonder what happens then. Are we restarting the querier?

Comment (ashwanthgoli) [reply pkg/querier/worker/processor_manager.go:68]: when more schedulers get added? which means each worker gets a smaller concurrency value.
not something that is often done in practice, so it should be alright I think

**Review by ashwanthgoli — APPROVED** [review]
lgtm

File: pkg/querier/worker/frontend_processor.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (fp *frontendProcessor) notifyShutdown(ctx context.Context, conn *grpc.Clie
 }
 
 // runOne loops, trying to establish a stream to the frontend to begin request processing.
-func (fp *frontendProcessor) processQueriesOnSingleStream(ctx context.Context, conn *grpc.ClientConn, address string) {
+func (fp *frontendProcessor) processQueriesOnSingleStream(ctx context.Context, conn *grpc.ClientConn, address, _ string) {
```
Comment (cstyan) [inline pkg/querier/worker/frontend_processor.go:61]: is the `_` param intentional here?

File: pkg/querier/worker/frontend_processor_test.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func TestRecvFailDoesntCancelProcess(t *testing.T) {
 		running.Store(true)
 		defer running.Store(false)
 
-		mgr.processQueriesOnSingleStream(ctx, cc, "test:12345")
+		mgr.processQueriesOnSingleStream(ctx, cc, "test:12345", "1")
```
Comment (cstyan) [inline pkg/querier/worker/frontend_processor_test.go:42]: is `"1"` necessary?

**Review by cstyan — COMMENTED** [review]
would the new native histograms be helpful here as well? these are much less costly and have less relative error between buckets

there's essentially a single series per native histogram metric and buckets are populated sparsely in storage

File: pkg/querier/worker/frontend_processor.go (hunk 2 above)
Comment (jeschkies) [reply pkg/querier/worker/frontend_processor.go:61]: Yes, `processQueriesOnSingleStream` implements the `processor` interface and we don't use the worker ID in the case of the frontend processor.

File: pkg/querier/worker/frontend_processor_test.go (hunk 3 above)
Comment (jeschkies) [reply pkg/querier/worker/frontend_processor_test.go:42]: You are right. Since it's ignored we don't need it here.

**Pull request merged**